h2/utilities.py

```diff
@@ -291,29 +295,101 @@ def _check_host_authority_header(headers, hdr_validation_flags):
     host_header_val = None
 
     for header in headers:
-        if header[0] == b':authority':
+        if header[0] in (b':authority', u':authority'):
```

Comment:
I’m sneaking this change in here, because it slipped through in #252, and it caused some tests to fail when I was playing with the sent header blocks. Okay? This mirrors what we do in `authority_from_headers`.